src/Snt/Capi/Http/ApiHttpClientConfiguration.php

```diff
@@ -39,6 +40,23 @@ public function __construct($endpoint, $apiKey, $apiSecret)
     }
 
     /**
+     * @param array $config
+     *
+     * @return ApiHttpClientConfiguration
+     * @throws InvalidArgumentException
+     */
+    public static function fromArray(array $config)
+    {
+        foreach (['endpoint', 'apiKey', 'apiSecret'] as $key) {
+            if (empty($config[$key])) {
+                throw new InvalidArgumentException(sprintf('Missing config parameter "%s"', $key));
```

Comment:
for me `MissingConfigurationParameterException` is still not so clear without exception message. I don't see another advantages with using custom exception, instead of general one. Can you provide some examples, when custom exception is better than general one?